Shadowblade_main.lua

```diff
@@ -0,0 +1,330 @@
+		nLastHarassUtil = nLastHarassUtil + (100 - nTargetHealth * 100)
+	end
+
+
+--]]
+
+	--Feint
+	if abilFeint:CanActivate() and bCanSee and unitSelf:GetMana() > 200 then
+		local nRange = abilFeint:GetRange()
+		if abilGargantuan:CanActivate() and nTargetDistanceSq < (nRange * nRange) then
+			bActionTaken = core.OrderAbilityEntity(botBrain, abilFeint, unitTarget)
+		end
+	end
+
+		--Gargantuan
+	if not bActionTaken and abilGargantuan:CanActivate() then
+		if nTargetDistanceSq < (300 * 300) then
+			bActionTaken = core.OrderAbility(botBrain, abilGargantuan)
+		--Soul is used only if Feint is on Cooldown, to be used in combo: Feint -> Gargantuan -> Soul
+			if abilFeint:CanActivate() then
+						bActionTaken = core.OrderAbilityEntity(botBrain, abilFeint, unitTarget)
+				elseif abilSoul:CanActivate() then
+						bActionTaken = core.OrderAbility(botBrain, abilSoul) 
+				end
+		end
+	end	
+		
+
+	if not bActionTaken then
+		return object.harassExecuteOld(botBrain)
+	end
+	
+end
+object.harassExecuteOld = behaviorLib.HarassHeroBehavior["Execute"]
+behaviorLib.HarassHeroBehavior["Execute"] = HarassHeroExecuteOverride
+
+-----------------------------
+--	 Retreat execute	 --
+-----------------------------
+
+--Modelled after Pure`Light's Magebane custom retreat code.
+--  this is a great function to override with using retreating skills, such as blinks, travels, stuns or slows.
+
+function behaviorLib.CustomRetreatExecute(botBrain)
+	bActionTaken = false
+
+	local unitSelf = core.unitSelf
+	local unitTarget = behaviorLib.heroTarget
+	if unitTarget == nil or not unitTarget:IsValid() then
+		return false
+	end
+
+	local vecMyPosition = unitSelf:GetPosition()       
+	local vecTargetPosition = unitTarget:GetPosition()
+	
+	local nTargetDistanceSq = Vector3.Distance2DSq(vecMyPosition, vecTargetPosition)
+	
+	--Counting the enemies
+	local tEnemies = core.localUnits["EnemyHeroes"]
+	local nCount = 0
+
+	local bCanSeeUnit = unitTarget and core.CanSeeUnit(botBrain, unitTarget)
+	for id, unitEnemy in pairs(tEnemies) do
+		if core.CanSeeUnit(botBrain, unitEnemy) then
+			nCount = nCount + 1
+		end
+	end
+
+	if unitSelf:GetHealthPercent() < .50 then
+		local abilGargantuan = skills.abilGargantuan
+		local abilEssence = skills.abilEssence
+		-- BACK OFF!
+		-- Use Gargantuan THEN Essence on SELF
+		if bCanSeeUnit and abilGargantuan:CanActivate() and abilEssence:CanActivate() then
+			bActionTaken = core.OrderAbility(botBrain, abilGargantuan)
+			bActionTaken = core.OrderAbility(botBrain, abilEssence, unitSelf)
+				
+				if bActionTaken then
+				object.bDefensiveGargantuan = true	-- Don't get aggressive if we're blinking away
+				
+				end
+		-- If Essence is on Cooldown, use only Gargantuan
+			elseif bCanSeeUnit and abilGargantuan:CanActivate() then
+			bActionTaken = core.OrderAbility(botBrain, abilGargantuan)
+				
+				if bActionTaken then
+				object.bDefensiveGargantuan = true	-- Don't get aggressive if we're blinking away
+				
+				end
```

Comment:
Minor indent errors.
Feint may also serve a use in attempting to escape, leap onto an ally/target closer to the pool.